src/main/java/de/wejago/mqtt2influx/service/MqttSubscriberService.java

```diff
@@ -1,33 +1,42 @@
 package de.wejago.mqtt2influx.service;
 
-import de.wejago.mqtt2influx.config.MqttProperties;
+import de.wejago.mqtt2influx.config.Device;
+import de.wejago.mqtt2influx.config.DevicesConfig;
+import de.wejago.mqtt2influx.factory.JsonSubscriberFactory;
 import jakarta.annotation.PostConstruct;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
 import org.eclipse.paho.client.mqttv3.IMqttClient;
+import org.eclipse.paho.client.mqttv3.IMqttMessageListener;
 import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
 import org.eclipse.paho.client.mqttv3.MqttException;
-
 import org.springframework.stereotype.Service;
 
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class MqttSubscriberService {
-
-    private final SensorMessageSubscriber sensorMessageSubscriber;
-
     private final IMqttClient mqttClient;
     private final MqttConnectOptions mqttConnectOptions;
-
-    private final MqttProperties mqttProperties;
+    private final DevicesConfig devicesConfig;
+    private final JsonSubscriberFactory jsonSubscriberFactory;
 
     @PostConstruct
     public void postConstruct() {
         try {
             mqttClient.connect(mqttConnectOptions);
-            mqttClient.subscribe(mqttProperties.getTopic(), sensorMessageSubscriber);
-            log.info("Successfully connected to MQTT client!");
+            for (Device device : devicesConfig.getDevices()) {
+                if (device != null && StringUtils.isNotBlank(device.getTopic())
```

Comment:
Move the check to an extra method with appropriate error messages.
Then a user would directly see which check fails and what he needs to fix in the configuration.